src/VisualStudio/Core/Def/Implementation/Remote/RemoteHostClientServiceFactory.SolutionChecksumUpdater.cs

```diff
@@ -122,6 +130,50 @@ private static void CancelAndDispose(CancellationTokenSource cancellationSource)
                 // dispose cancellation token source
                 cancellationSource.Dispose();
             }
+
+            private void PushTextChanges(Document oldDocument, Document newDocument)
+            {
+                SourceText oldText = null;
+                SourceText newText = null;
+                if ((oldDocument?.TryGetText(out oldText) == false) ||
+                    (newDocument?.TryGetText(out newText) == false))
+                {
+                    // we only support case where text already exist
+                    return;
```

Comment:
if this returns... what does that mean for the caller?  They don't seem to know that no changes were pushed.  Will that leave us in an inconsistent state?  Should we instead bail out and push the entire change?

Reply:
nobody should care whether this succeeded or not except OOP engine. this will improve perf in one of major scenario (typing) by not moving whole content to the other side. but whether that happened or not doesn't impact features functionality.